src/planner/filter/bloom_filter.cpp

```diff
@@ -134,10 +135,66 @@ bool BFTableFilter::FilterValue(const Value &value) const {
 	return filter.LookupOne(hash);
 }
 
-FilterPropagateResult BFTableFilter::CheckStatistics(BaseStatistics &stats) const {
+template <class T>
+static FilterPropagateResult TemplatedCheckStatistics(const BloomFilter &bf, const BaseStatistics &stats) {
+	if (!NumericStats::HasMinMax(stats)) {
+		return FilterPropagateResult::NO_PRUNING_POSSIBLE;
+	}
+
+	const auto min = NumericStats::GetMin<T>(stats);
+	const auto max = NumericStats::GetMax<T>(stats);
+	if (min > max) {
+		return FilterPropagateResult::NO_PRUNING_POSSIBLE; // Invalid stats
+	}
+	T range_typed;
+	if (!TrySubtractOperator::Operation(max, min, range_typed) || range_typed > 2048) {
+		return FilterPropagateResult::NO_PRUNING_POSSIBLE; // Overflow or too wide of a range
+	}
+	const auto range = NumericCast<idx_t>(range_typed);
+
+	T val = min;
+	idx_t hits = 0;
+	for (idx_t i = 0; i <= range; i++) {
+		hits += bf.LookupOne(Hash(val));
```

Comment:
nit: maybe just do `Hash(min + i)` to avoid having to do the conditional incrementing of `val`?